webapp/src/app/shared/widgets/number-widget/number-widget.component.ts

```diff
@@ -1,14 +1,24 @@
-import {Component, OnInit} from '@angular/core';
+import {Component, OnInit, ViewChild, ElementRef} from '@angular/core';
 import {AbstractWidget} from '../abstract-widget';
 
 @Component({
 	selector: 'app-number-widget',
 	templateUrl: './number-widget.component.html',
 	styleUrls: ['./number-widget.component.scss', '../widget.scss']
 })
-export class NumberWidgetComponent extends AbstractWidget {
-	constructor() {
-		super();
-	}
+export class NumberWidgetComponent extends AbstractWidget implements OnInit {
+
+	setSettings(key: string, value: string) {
```

Comment:
this function name is easily mistaken for the other setSetting function. It would make more sense to give it a completely different name or just override the setSetting method